lib/Language/languageHelpers.ts

```diff
@@ -1,18 +1,35 @@
-import i18next from "i18next";
+import { i18n } from "i18next";
+import { isJsonString } from "../Core/Json";
+
+export const TRANSLATE_KEY_PREFIX = "translate#";
 
 /**
- * Takes a given string and translates it if it exists, otherwise return
+ * Takes a given string and translates it if it's a translation key that exists,
+ * otherwise returns the string.
+ *
+ * This function doesn't use the global i18next instance because this function is predominately
+ * used in the UIMode and using React components should use withTranslation HOC or useTranslation
+ * hook to ensure that they update when the aop language changes.
+ *
+ * @param keyOrString Either a prefixed translation key or a string
+ * @param i18n An i18next instance
  */
-export function useTranslationIfExists(
-  keyOrString: string = "",
+export function applyTranslationIfExists(
+  keyOrString: string,
+  i18n: i18n,
   options?: { [key: string]: string }
-) {
-  if (keyOrString.indexOf("translate#") === 0) {
-    const translationKey = keyOrString.substr("translate#".length);
-    return i18next.exists(translationKey)
-      ? i18next.t(translationKey, options)
-      : translationKey;
-  } else {
-    return keyOrString;
+): string {
+  // keyOrString could be undefined in some cases even if we type it as string
+  if (isJsonString(keyOrString as unknown)) {
```

Comment:
@nf-s I am more into this change as it will force us to use string type and in the future values in config will be part of the traits system and validated as a string, and non-string value shouldn't come from other places

Reply:
Yes that what I was thinking @zoran995, if you see my previous commit I typed it as unknown, but then thought better to type as string so we are forced to pass a string in ost occurrences, and if a non-string value sneaks through we handle it inside `applyTranslationIfExists()`. Happy to defer to you @nf-s but do you think that intention is valid? Perhaps my type assertion in `if (isJsonString(keyOrString as unknown))` is not the correct way to do things.

Reply:
Ok happy to leave it then

Reply:
In my latest debugging, I think `name` and `title` doesn't get used anywhere. Will clean it up sometime - https://github.com/TerriaJS/terriajs/issues/6368